camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -222,16 +232,11 @@ public void run() throws Exception
       {
         System.err.println("unable to process " + topicName + " skipping...");
         e.printStackTrace();
-        executorService.shutdown();
-        throw e;
       }
     }
 
-    if (!executorService.isTerminated()) {
-      log.info("Shutting down priority executor");
-      executorService.shutdown();
-    }
-
+    log.info("Shutting down priority executor");
+    executorService.shutdown();
     while (!executorService.isTerminated())
     {
       executorService.awaitTermination(30, TimeUnit.SECONDS);
```

Comment:
Isn't 30 seconds  too short to wait for all topics and all paths? (they all happen async as there's no blocking inside `runCollectorForTopicDir` anymore)